python/taichi/lang/ast/ast_transformer.py

```diff
@@ -1145,6 +1146,19 @@ def build_static_for(ctx, node, is_grouped):
             if len(targets) != 1:
                 raise TaichiSyntaxError(f"Group for should have 1 loop target, found {len(targets)}")
             target = targets[0]
+            iter_time = 0
+            if ti_unroll_limit:
+                for value in impl.grouped(ndrange_arg):
+                    iter_time += 1
+                    if iter_time > ti_unroll_limit:
+                        warnings.warn_explicit(
+                            f"loop unrolling only support under {ti_unroll_limit}",
```

Comment:
Maybe a better warning message: `You are unrolling more than {ti_unroll_limit} iterations, so the compile time may be extremely long. You can use a non-static for loop if you want to decrease the compile time. You can disable this warning by setting ti.init(unrolling_limit=0).`